rs/bitcoin/adapter/src/common.rs

```diff
@@ -16,6 +22,182 @@ pub const DEFAULT_CHANNEL_BUFFER_SIZE: usize = 64;
 /// This field contains the datatype used to store height of a Bitcoin block
 pub type BlockHeight = u32;
 
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+#[allow(missing_docs)]
+/// AdapterNetwork selects between bitcoin and dogecoin Network.
+///
+/// The string representation for mainnet would would be either "bitcoin" or "dogecoin".
+/// But for non-mainnet networks, they would have a prefix of either "bitcoin:" or "dogecoin:".
+///
+/// The parsing from string on the other hand favors bitcoin network in order
+/// to maintain backward compatibility. It is only when a string fails to parse
+/// as a bitcoin network, it will try to parse as a dogecoin network (with prefix "dogcoin:").
```

Comment:
```suggestion
/// The parsing from string on the other hand favors Bitcoin network in order
/// to maintain backward compatibility. It is only when a string fails to parse
/// as a Bitcoin network, it will try to parse as a Dogecoin network (with prefix "dogcoin:").
```